cpp/src/phonenumbers/base/thread_checker.h

```diff
@@ -22,16 +22,18 @@
 // Note that I18N_PHONENUMBERS_NO_THREAD_SAFETY must be defined only to let the
 // user of the library know that it can't be used in a thread-safe manner when
 // it is not depending on Boost.
-#if !defined(__linux__) && !defined(__APPLE__) && \
-    !defined(I18N_PHONENUMBERS_NO_THREAD_SAFETY)
+#if !defined(__linux__) && !defined(__APPLE__) && !defined(PHONENUMBERS_HAVE_POSIX_THREAD) && \
+    !defined(I18N_PHONENUMBERS_NO_THREAD_SAFETY) && \
+	!((__cplusplus>=201103L) && (!(defined(NO_LIBPHONENUMBERS_STDMUTEX)))) && \
```

Comment:
nit: spaces around >=
nit: please replace (!( with !

If you want to support NO_LIBPHONENUMBERS_STDMUTEX, please add it to lock.h, singleton.h and CMakeLists.txt.